### PR TITLE
Full body response on onError option for spf.navigate

### PR DESCRIPTION
**Pull request by tchiotludo**
For now, we just received on simple detail.err string for failed navigate on option onError, I've added a detail.xhr with the full XMLHttpRequest in order to have more detail and responseText for debuging or displaying

### CODE REVIEW

File: src/client/nav/nav.js [hunk 1]
```diff
@@ -673,14 +673,15 @@ spf.nav.navigateAddHistory_ = function(url, referer, handleError) {
  * @param {spf.RequestOptions} options Request options object.
  * @param {string} url The requested URL, without the SPF identifier.
  * @param {Error} err The Error object.
+ * @param {XMLHttpRequest=} xhr The XMLHttpRequest for current error
  * @private
  */
-spf.nav.handleNavigateError_ = function(options, url, err) {
+spf.nav.handleNavigateError_ = function(options, url, err, xhr) {
```
Comment (DavidCPhillips) [inline src/client/nav/nav.js]: Nit: Optional fields should be named opt_*

File: src/client/nav/nav.js [hunk 2]
```diff
@@ -1305,11 +1306,12 @@ spf.nav.process = function(response, opt_callback) {
  * @param {?spf.RequestOptions=} opt_options Optional request options object.
  * @param {boolean=} opt_noEvents Whether to skip the event and only execute the
  *     callback; for use with load and prefetch requests.
+ * @param {XMLHttpRequest=} xhr The XMLHttpRequest for current error
  * @return {boolean} False if the event was canceled.
  * @private
  */
-spf.nav.dispatchError_ = function(url, err, opt_options, opt_noEvents) {
-  var detail = {'url': url, 'err': err};
+spf.nav.dispatchError_ = function(url, err, opt_options, opt_noEvents, xhr) {
```
Comment (DavidCPhillips) [inline src/client/nav/nav.js]: Ditto: opt_*

**Comment by DavidCPhillips** [comment]
Whoops, this one slipped a bit.

The changes look good. Could you squash the commits into one? Then we'll go ahead and merge it in.

**Comment by tchiotludo** [comment]
it's done.
Thanks

**Comment by DavidCPhillips** [comment]
LGTM

**Pull request merged**